python/dgl/backend/mxnet/tensor.py

```diff
@@ -368,28 +368,57 @@ def forward(self, lhs_data, rhs_data):
                             ctx=lhs_data.context, dtype=lhs_data.dtype)
         out_data_nd = zerocopy_to_dgl_ndarray_for_write(out_data)
         K.binary_op_reduce(
-            self.reducer, self.binary_op, self.graph, self.lhs, self.rhs,
+            self.reducer if self.reducer != 'mean' else 'sum',
+            self.binary_op, self.graph, self.lhs, self.rhs,
             lhs_data_nd, rhs_data_nd, out_data_nd, self.lhs_map[0],
             self.rhs_map[0], self.out_map[0])
+        # normalize if mean reducer
```

Comment:
Could you leave a note that this is a temporary hack and we should have a better solution (either in kernel or to support int32 queries) in the future?

Reply:
DONE